link/src/main/java/com/stripe/android/link/ui/signup/SignUpScreenState.kt

```diff
@@ -0,0 +1,18 @@
+package com.stripe.android.link.ui.signup
+
+import com.stripe.android.core.strings.ResolvableString
+import com.stripe.android.uicore.elements.PhoneNumberController
+import com.stripe.android.uicore.elements.TextFieldController
+
+sealed interface SignUpScreenState {
+    data class Content(
+        val emailController: TextFieldController,
+        val phoneNumberController: PhoneNumberController,
+        val nameController: TextFieldController,
+        val signUpEnabled: Boolean,
+        val signUpState: SignUpState = SignUpState.InputtingPrimaryField,
+        val errorMessage: ResolvableString? = null
+    ) : SignUpScreenState
+
+    data object Loading : SignUpScreenState
```

Comment:
Why is this needed? I would expect we'd have a loading state for Link as a whole, and then each individual screen shouldn't need to have a loading state.

If there's specific data we are waiting on, do we need the whole screen to be in a loading state? Or can we either:
- wait to transition to the screen until we have the info we need or 
- keep the specific UI that's dependent on the async state in a loading state, rather than keeping the whole UI in a loading state

Reply:
I took this out. The original link implementation checked for log in state on both the LinkViewModel and SignUpViewModel, but you're right, it shouldn't be duplicated.